bitswap.go

```diff
@@ -359,15 +348,13 @@ func (bs *Bitswap) receiveBlockFrom(blk blocks.Block, from peer.ID) error {
 
 // SessionsForBlock returns a slice of all sessions that may be interested in the given cid
 func (bs *Bitswap) SessionsForBlock(c cid.Cid) []*Session {
-	bs.sessLk.Lock()
-	defer bs.sessLk.Unlock()
-
 	var out []*Session
-	for _, s := range bs.sessions {
+	bs.sm.IterateSessions(func(session exchange.Fetcher) {
+		s := session.(*Session)
```

Comment:
Having this cast is a bit funky. I get that this is to avoid a dependency issue but I wonder if there's a better way. Would it make sense to move *sessions* into their own package? That way, SessionManager could operate over concrete sessions.

Unfortunately, I don't think that'll work as Session appears to need access to bitswap internals. An alternative is to create a `Session` interface that exposes an `InterestedIn` method.

I'd be fine punting these questions till later.

Reply:
Hey!

Let's punt till later since two PR's forward Sessions end up in their own package, and this cast is gone :)